Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4784,6 +4785,46 @@ void WebPageProxy::disconnectFramesFromPage()
         mainFrame->webProcessWillShutDown();
 }
 
+void WebPageProxy::disownOpener()
+{
+    if (!m_process->processPool().configuration().processSwapsOnWindowOpenWithOpener())
+        return;
+    if (m_initialWindow && m_opener) {
+        auto process = WebProcessProxy::processForIdentifier(m_initialWindow->processIdentifier);
+        process->send(Messages::WebProcess::RemoveRemoteWindow(*m_opener), 0);
+    }
+    m_opener = std::nullopt;
+}
+
+void WebPageProxy::didCreateWindow(uint64_t frameID, GlobalWindowIdentifier&& windowIdentifier, std::optional<GlobalWindowIdentifier> openerID)
+{
+    if (!m_process->processPool().configuration().processSwapsOnWindowOpenWithOpener())
+        return;
+    if (!m_mainFrame || m_mainFrame->frameID() != frameID)
+        return;
+    if (!m_initialWindow)
+        m_initialWindow = windowIdentifier;
+
+    if (openerID)
+        m_opener = openerID;
+
+    if (m_opener && m_initialWindow && windowIdentifier.processIdentifier != m_opener->processIdentifier) {
```

Comment:
I understand you only care about window openers in this patch but I think we should try and be a little bit more forward-looking and try to be a little bit more generic. This code seems very specific to opener and it is not clear to me why it needs to be.